integration/src/main/java/scripts/utils/ClientTestUtils.java

```diff
@@ -338,6 +341,76 @@ public static <T> T getWithRetryOnException(
     return result;
   }
 
+  /**
+   * Get a result from a call that might throw an exception. If the supplier finishes, the result is
```

Comment:
update javadoc - this method will only return if the supplier finishes and the supplied value satisfies the predicate